packages/cli/src/commands/init/index.ts

```diff
@@ -47,5 +47,10 @@ export default {
       description:
         'Inits a project with a custom package name (Android) and bundle ID (iOS), e.g. com.example.app',
     },
+    {
+      name: '--platform-name <string>',
+      description:
+        'Name of out of tree platform to be used for ex. react-native-macos. This flag is optional as it should be passed automatically by out of tree platform. It needs to match the name of the platform declared in package.json',
```

Comment:
FYI @saadnajmi

Note that macOS and Windows both have separate init packages right now (with the windows one in particular growing quite complicated). I worry naming react-native-macOS here might be jumping ahead a bit, until that init flow has changed (unless it already has).

Reply:
The init flow stays the same. While working on a new platform (visionos) we want to make things right and easy for any other platform to follow. Today this particular flag is quite virtual, as it's passed from the programmatic CLI usage in core. It's still in flux and once we're more firm on the design, we'll likely contribute some PoC to macOS and hopefully Windows too.